yrs/src/block.rs

```diff
@@ -808,7 +855,7 @@ impl Item {
         let clock = self.id.clock;
         let other = Item {
             id: ID::new(client, clock + diff),
-            left: Some(BlockPtr::from(ID::new(client, clock + diff - 1))),
+            left: Some(BlockPtr::from(self.id)),
```

Comment:
Again - while left origin points to the end of a previous block, left pointer point to the beginning of it.